chain/bitcoind_events.go

```diff
@@ -268,6 +292,11 @@ func (b *bitcoindZMQEvents) blockEventHandler() {
 				return
 			}
 
+			// From our local mempool map, let's remove each of the
+			// transactions that are confirmed in this new block,
+			// since they are no longer in the mempool.
+			b.mempool.clean(block.Transactions)
```

Comment:
Yeah good point, I think we've seen this duplicate tx notifications elsewhere as well. It's also the case that we _do_ handle blocks vs txns in diff goroutines as well. I think this is why we went with the approach of marked vs unmarked earlier as well?

Reply:
Cool now updated the logic to take advantage of the marks. And this is now what happens,
- when a new block comes in, it will first delete all the unmarked transactions from the previous round.
- if it read block notification first, then it unmarks all the transactions from the block. When the same tx is received in rawtx again, it won't be processed.